tests/basic/storage_pool_test.go

```diff
@@ -8282,3 +8282,106 @@ var _ = Describe("{AddDiskAddDriveAndDeleteInstance}", func() {
+
+	*/
+	var testrailID = 0
+	// Testrail Description : Add drive when as journal
+	var runID int
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("DriveAddAsJournal",
+			"Add drive when as journal",
+			nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+	var contexts []*scheduler.Context
+	stepLog := "Add drive when as journal"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("adddriveasjournal-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+
+		// Get Pool with running IO on the cluster
+		poolUUID, err := GetPoolIDWithIOs(contexts)
+		log.FailOnError(err, "Failed to get pool running with IO")
+		log.InfoD("Pool UUID on which IO is running [%s]", poolUUID)
+
+		// Get Node Details of the Pool with IO
+		nodeDetail, err := GetNodeWithGivenPoolID(poolUUID)
+		log.FailOnError(err, "Failed to get Node Details from PoolUUID [%v]", poolUUID)
+		log.InfoD("Pool with UUID [%v] present in Node [%v]", poolUUID, nodeDetail.Name)
+		// Get PX StorageCluster
+		var dmthinEnabled bool
+		cluster, err := Inst().V.GetDriver()
+		if err != nil {
+			log.FailOnError(err, "Error getting clusterspecs")
+		}
+		argsList, err := util.MiscArgs(cluster)
+		for _, args := range argsList {
+			if strings.Contains(args, "dmthin") {
+				dmthinEnabled = true
+			}
+		}
+
+		// Add cloud drive on the node selected and wait for rebalance to happen
+		driveSpecs, err := GetCloudDriveDeviceSpecs()
+		log.FailOnError(err, "Error getting cloud drive specs")
+
+		deviceSpec := driveSpecs[0]
+		devicespecjournal := deviceSpec + " --journal"
+		if dmthinEnabled {
+			err := Inst().V.AddCloudDrive(nodeDetail, devicespecjournal, -1)
+			re := regexp.MustCompile(".*Journal/Metadata device add not supported for PX-StoreV2*")
+			dash.VerifyFatal(re.MatchString(fmt.Sprintf("%v", err)),
+				false,
+				"Failed to match the error while adding drive")
+			log.InfoD(fmt.Sprintf("Errored while adding Pool as expected on Node [%v]", nodeDetail.Name))
+		} else {
+			err = Inst().V.EnterPoolMaintenance(*nodeDetail)
+			log.InfoD("Enter pool Maintenance mode ")
+			isjournal, err := isJournalEnabled()
+			if isjournal {
+				devicespecjournal := deviceSpec + " --journal"
+				err = Inst().V.AddCloudDrive(nodeDetail, devicespecjournal, -1)
+				re := regexp.MustCompile(".*journal exists*")
+				dash.VerifyFatal(re.MatchString(fmt.Sprintf("%v", err)),
+					false,
+					"Failed to match the error while adding drive")
+				log.InfoD(fmt.Sprintf("Errored while adding Pool as expected on Node [%v]", nodeDetail.Name))
+			} else {
+				devicespecjournal := deviceSpec + " --journal"
+				err = Inst().V.AddCloudDrive(nodeDetail, devicespecjournal, -1)
+				if err != nil {
+					log.FailOnError(err, "journal add failed")
+				} else {
```

Comment:
You don't need to wrap `log.FailOnError(err, "journal add failed")` inside `if err != nil {}`, it will already error out if the error is not `nil`

Reply:
Done